src/riak_core_info_service.erl

```diff
@@ -110,7 +110,27 @@
                            {error, term()}.
 
 start_service(Registration, Shutdown, Provider, Handler) ->
-    riak_core_info_service_sup:start_service(Registration, Shutdown, Provider, Handler).
+    case verify_callable([Registration, Shutdown, Handler]) of
+        true ->
+            riak_core_info_service_sup:start_service(Registration, Shutdown, Provider, Handler);
+        {false, {_Module, _Function, _Arity}=Fun} ->
+            {error, {not_callable, Fun}}
+    end.
+
+verify_callable([]) ->
```

Comment:
I'm happier with custom recursion in cases where we fail early, although admittedly there are only 3 functions to check.